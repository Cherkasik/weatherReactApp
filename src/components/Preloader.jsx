import React from 'react';
import styled from 'styled-components';
import { ReactComponent as Refresh } from '../assets/Refresh.svg'

const Container = styled.div`
  width: 100%;
  text-align: center;
  margin-top: 20px;
`;

const Text = styled.div`
  font-size: ${props => props.small ? '25px' : '50pt'};
  text-align: center;
  color: #6F8185;
`;

const RefreshButton = styled.button`
  width: 250px;
  height: 80px;
  margin-top: 20px;
  background-color: #C9D2D7;
  border: none;
  svg {
    width: 50px;
    height: 50px;
    fill: #6F8185;
  }
`;

function Preloader(props) {
  const { small, repeat, error } = props;
  return (
    <Container>
      <Text small={small}>{error ? error : 'Подождите, данные загружаются'}</Text>
      <RefreshButton small={small} onClick={repeat}><Refresh /></RefreshButton>
    </Container>
  );
}

export default Preloader;
