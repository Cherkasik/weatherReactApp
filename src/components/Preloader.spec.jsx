import React from 'react';
import 'jest-styled-components';
import { render } from '@testing-library/react';
import Preloader from './Preloader';

describe('Preloader component', () => {
  const repeat = jest.fn();
  const error = 'Some error';
  it('should render small with error', () => {
    const { container } = render(
      <Preloader
        repeat={repeat}
        small
        error={error}
      />
      );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render big with error', () => {
    const { container } = render(
      <Preloader
        repeat={repeat}
        error={error}
      />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render small without error', () => {
    const { container } = render(
      <Preloader
        repeat={repeat}
        small
      />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render big without error', () => {
    const { container } = render(
      <Preloader
        repeat={repeat}
      />
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});