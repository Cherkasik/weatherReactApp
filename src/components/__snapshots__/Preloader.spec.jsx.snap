// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Preloader component should render big with error 1`] = `
.c0 {
  width: 100%;
  text-align: center;
  margin-top: 20px;
}

.c1 {
  font-size: 50pt;
  text-align: center;
  color: #6F8185;
}

.c2 {
  width: 250px;
  height: 80px;
  margin-top: 20px;
  background-color: #C9D2D7;
  border: none;
}

.c2 svg {
  width: 50px;
  height: 50px;
  fill: #6F8185;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    Some error
  </div>
  <button
    class="c2"
  >
    <svg>
      Refresh.svg
    </svg>
  </button>
</div>
`;

exports[`Preloader component should render big without error 1`] = `
.c0 {
  width: 100%;
  text-align: center;
  margin-top: 20px;
}

.c1 {
  font-size: 50pt;
  text-align: center;
  color: #6F8185;
}

.c2 {
  width: 250px;
  height: 80px;
  margin-top: 20px;
  background-color: #C9D2D7;
  border: none;
}

.c2 svg {
  width: 50px;
  height: 50px;
  fill: #6F8185;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    Подождите, данные загружаются
  </div>
  <button
    class="c2"
  >
    <svg>
      Refresh.svg
    </svg>
  </button>
</div>
`;

exports[`Preloader component should render small with error 1`] = `
.c0 {
  width: 100%;
  text-align: center;
  margin-top: 20px;
}

.c1 {
  font-size: 25px;
  text-align: center;
  color: #6F8185;
}

.c2 {
  width: 250px;
  height: 80px;
  margin-top: 20px;
  background-color: #C9D2D7;
  border: none;
}

.c2 svg {
  width: 50px;
  height: 50px;
  fill: #6F8185;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    Some error
  </div>
  <button
    class="c2"
  >
    <svg>
      Refresh.svg
    </svg>
  </button>
</div>
`;

exports[`Preloader component should render small without error 1`] = `
.c0 {
  width: 100%;
  text-align: center;
  margin-top: 20px;
}

.c1 {
  font-size: 25px;
  text-align: center;
  color: #6F8185;
}

.c2 {
  width: 250px;
  height: 80px;
  margin-top: 20px;
  background-color: #C9D2D7;
  border: none;
}

.c2 svg {
  width: 50px;
  height: 50px;
  fill: #6F8185;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    Подождите, данные загружаются
  </div>
  <button
    class="c2"
  >
    <svg>
      Refresh.svg
    </svg>
  </button>
</div>
`;
